### ISSUE TITLE
on_publish_diagnostics not respected

### ISSUE THREAD

**Issue by ranebrown** [jose-elias-alvarez/null-ls.nvim#356]
### FAQ

- [X] I have checked the [FAQ](https://github.com/jose-elias-alvarez/null-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

v0.6.0-dev+2ef9d2a66

### null-ls config

```
vim.lsp.handlers["textDocument/publishDiagnostics"] = vim.lsp.with(
    vim.lsp.diagnostic.on_publish_diagnostics,
    {
        underline = false,
        virtual_text = false,
        signs = true,
        update_in_insert = false,
    }
)

    local sources = {
        null_ls.builtins.formatting.stylua.with({
            condition = function()
                return vim.fn.executable("stylua") > 0
            end,
        }),
        null_ls.builtins.diagnostics.markdownlint.with({
            filetypes = { "markdown", "vimwiki" },
            condition = function()
                return vim.fn.executable("markdownlint") > 0
            end,
        }),
        null_ls.builtins.formatting.black.with({
            condition = function()
                return vim.fn.executable("black") > 0
            end,
        }),
        null_ls.builtins.formatting.isort.with({
            condition = function()
                return vim.fn.executable("isort") > 0
            end,
        }),
        null_ls.builtins.diagnostics.flake8.with({
            extra_args = { "--max-line-length", "88" },
            condition = function()
                return vim.fn.executable("flake8") > 0
            end,
        }),
        null_ls.builtins.diagnostics.hadolint.with({
            condition = function()
                return vim.fn.executable("hadolint") > 0
            end,
        }),
        null_ls.builtins.formatting.shfmt.with({
            extra_args = { "-i", "4", "-ci" },
            condition = function()
                return vim.fn.executable("shfmt") > 0
            end,
        }),
        null_ls.builtins.diagnostics.shellcheck.with({
            condition = function()
                return vim.fn.executable("shellckeck") > 0
            end,
        }),
        null_ls.builtins.diagnostics.vint.with({
            condition = function()
                return vim.fn.executable("vint") > 0
            end,
        }),
        null_ls.builtins.formatting.prettier.with({
            filetypes = { "json", "yaml", "markdown", "vimwiki", "xml" },
            extra_args = function(params)
                if params.ft == "markdown" or params.ft == "vimwiki" then
                    return { "--tab-width", "4" }
                elseif params.ft == "xml" then
                    return { "--parser", "xml" }
                end
            end,
            condition = function()
                return vim.fn.executable("prettier") > 0
            end,
        }),
    }
    null_ls.config({
        diagnostics_format = "#{s}: #{m} [#{c}]",
        sources = sources,
    })
    local config = make_config()
    lspconfig["null-ls"].setup(config)
```

### Steps to reproduce

Checkout commit 0d10c1f7d9f071ba4d49652c275a3f85665c0368 and virtual text is disabled. Checkout any newer commit and it is not disabled.

### Expected behavior

Setting `vim.lsp.diagnostic.on_publish_diagnostics` with `virtual_text = false,` disables virtual text diagnostic messages.

### Actual behavior

Sometime after 0d10c1f7d9f071ba4d49652c275a3f85665c0368 disabling virtual text for diagnostic messages no longer works.

### Debug log

Let me know if posting the log would be useful. Didn't see anything relevant. 

### Help

Yes, but I don't know how to start. I would need guidance

### Implementation help

Understand reason why the handler isn't being used and pointed to the correct section of code.

**Comment by jose-elias-alvarez** [comment]
We're now using the new diagnostic API when it's available (0.6.0), meaning that you'll want to migrate to `vim.diagnostic.config`. The options are a little bit different, but in your case I think you can change your handler to this:

```lua
vim.diagnostic.config({
    underline = false,
    virtual_text = false,
    signs = true,
    update_in_insert = false,
})
```

**Comment by ranebrown** [comment]
Thank you!
